pkg/segment/reader/segread/multicolreader.go

```diff
@@ -353,11 +351,13 @@ func (mcsr *MultiColSegmentReader) ExtractValueFromColumnFile(colKeyIndex int, b
 }
 
 func (mcsr *MultiColSegmentReader) returnBuffers() {
-
 	if mcsr.allFileReaders != nil {
 		for _, reader := range mcsr.allFileReaders {
 			if reader != nil {
-				reader.ReturnBuffers()
+				err := reader.ReturnBuffers()
+				if err != nil {
+					log.Errorf("returnBuffers: Error returning buffer back to memory: %v", err)
```

Comment:
nit: I would put `MultiColSegmentReader.returnBuffers: ...` as the log line to better more clarity. But you can keep what you have